pennylane/ops/qubit/attributes.py

```diff
@@ -75,6 +75,13 @@ def __contains__(self, obj):
             return super().__contains__(obj)
 
         try:
+
+            # Hotfix: return False for all tensors.
+            # Can be removed or updated when tensor class is
+            # improved.
+            if isinstance(obj, Tensor):
+                return False
+
```

Comment:
It turns out that the adjoint method feeds tensor observables to the ``apply_operation`` method :) 

This is by far the easiest fix for now, and if we want to add tensors to attributes we can revisit!

Reply:
Oh, the reason why `tensor in attribute` fails is that it tries to check if a set contains a list (the name of tensors is a list of names, which is not ideal).

Reply:
I cannot remember, are we keeping the `Tensor` class after the overhaul?

Reply:
Wait

What happens if you want to do
```python
qml.PauliZ(0) @ qml.PauliZ(1) in diagonal_in_z_basis
```

🤔 

Can we support this somehow?

> the name of tensors is a list of names, which is not ideal

I've always very much disliked this!

> It turns out that the adjoint method feeds tensor observables to the apply_operation method :)

the `Device.apply_operation` method? I didn't think this was used by any qubit devices any more?